### DISCUSSION TITLE
Sengled Bulbs: Missing Power/Energy Information

### DISCUSSION THREAD

**Discussion by StephenJondrew** [Koenkk/zigbee2mqtt#13302] [Devices]
I have multiple Sengled E11-N1EA bulbs. Although they work correctly for operation under zigbee2mqtt, it appears that they don't include Power/Energy information. Previously I have connected these bulbs before to Home Assistant using ZHA and the bulbs provide two additional fields - Power (Wattage) and Energy (kWh). 

What is the best way to look to getting these features added to zigbee2mqtt support? If this is not the best place to ask, please let me know where I should post.

**Comment by Koenkk** [comment]
Could you check if the issue is fixed with the following external converter:

```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['E11-N1EA'],
    model: 'E11-N1EA',
    vendor: 'Sengled',
    description: 'Element plus color (A19)',
    fromZigbee: extend.light_onoff_brightness().fromZigbee.concat([fz.metering]),
    toZigbee: extend.light_onoff_brightness().toZigbee,
    ota: ota.zigbeeOTA,
    configure: async (device, coordinatorEndpoint, logger) => {
        await extend.light_onoff_brightness().configure(device, coordinatorEndpoint, logger);
        device.powerSource = 'Mains (single phase)';
        device.save();

        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'seMetering']);
        await reporting.readMeteringMultiplierDivisor(endpoint);
        await reporting.currentSummDelivered(endpoint);
        await reporting.instantaneousDemand(endpoint);
    },
    exposes: extend.light_onoff_brightness().exposes.concat([e.power(), e.energy()]),
};

module.exports = definition;

```

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m, check if issue is fixed

**Reply by StephenJondrew** [reply]
Thank you for the prompt reply - that worked perfectly! Much appreciated.

**Reply by Koenkk** [reply]
Great, this will be included in the next release on 1 August.

**Reply by StephenJondrew** [reply]
Fantastic to hear - thank you! Also, I know someone who is using ZHA with the non-colour version of these bulbs (E11-G13), and they advise that they also have Energy/Watt consumption. I do not own any of this model so I can not validate that this would also work for those bulbs, but I thought I'd pass it along.

**Reply by Koenkk** [reply]
z2m already supports it for the E11-G13: https://www.zigbee2mqtt.io/devices/E11-G13.html#power-numeric

**Reply by StephenJondrew** [reply]
Oh perfect - I didn't realize that. Thanks again for the help.

**Reply by StephenJondrew** [reply]
Hello. I didn't see this in the release notes for the 1st - was curious if it was added. Just wanted to double check before I removed the external converter. :)

Thanks again for your help.

**Reply by Koenkk** [reply]
Yes it is in (and also in the release notes of 1.27.0)

![image](https://user-images.githubusercontent.com/2892853/182406945-c8d32e64-61ec-482e-9075-3670089dbe26.png)

**Reply by StephenJondrew** [reply]
I must have missed it in the notes - thank you!

**Comment by russellmoran99** [comment]
I changed my zigbee channel today and had to repair all of my devices.  I no longer have color options.  I have followed the instructions provided by @Koenkk and created the ext_converter.js.  I am still not able to show the colors.  Any helps appreciated.
